### PR TITLE
Llama 3 Models

### PR DESCRIPTION
**Pull request by ym1234**

### CODE REVIEW

**Comment by geohot** [comment]
Bounty locked.

They changes the tokenizer?

Can you test the 70B model?

**Comment by ym1234** [comment]
> They changed the tokenizer?

Yeah, the actual model changes are pretty minimal

> Can you test the 70B model?

No I don't have enough memory

**Comment by ym1234** [comment]
> > They changed the tokenizer?
> 
> Yeah, the actual model changes are pretty minimal

Just to clarify, I diffed model.py in both repos and aside from comments, and one minor change (ParallelEmbeddings was changed to VocabParallelEmebeddings which i think just masks the tensor for each gpu instead of sharding it)  they are exactly the same. Only the tokenizer change is needed for full support in tinygrad.

**Comment by geohot** [comment]
I'll test 70B tomorrow, if it's good, bounty is yours.

File: examples/llama.py [hunk 1]
```diff
@@ -452,37 +452,23 @@ def main(string: str):
...
   # chatbot loop
   while 1:
     # add tokens from user in chatbot mode
     if chatbot:
       user_prompt = user_delim + input(user_delim) + "\n"
       outputted += user_prompt
+      toks.extend(llama.tokenizer.encode(user_prompt))
 
-    new_toks = [llama.tokenizer.bos_id()] + llama.tokenizer.encode(outputted)
-    assert toks == new_toks[:len(toks)]
```
Comment (ym1234) [inline examples/llama.py:416]: This assert doesn't hold anymore, for example `]\n` encodes to 933 but `]\n\n` is `2595` so concating `]\n` and `\n` changes the result.

**Comment by geohot** [comment]
Assuming benchmark tests pass on master, bounty is yours!

Can pay via PayPal or USDC, reach out to george@tinygrad.org to claim

**Comment by ym1234** [comment]
Sent

**Pull request merged**